### ISSUE TITLE
make "tiles" arbitrary polygons

### ISSUE THREAD

**Issue by mikelmaron** [hotosm/osm-tasking-manager#142]
add ability to define job as a set of arbitrary geometries, rather than automatically tile. 

thinking on approach would be to add geometry column to Tile model, and to_polygon would check this column. x/y/zoom would still be set, only iterating x, since it's used as a tile id.

an alternate "TileBuilder" would be developed, that reads in a list of geometries, and creates tasks. interface for this would be upload of geojson file.

would this more or less do it, or is there other logic that needs modification?

**Comment by mikelmaron** [comment]
Pierre, how is this possibly related to the Freehand branch? https://github.com/hotosm/osm-tasking-manager/tree/freehand

**Comment by pgiraud** [comment]
Yep, this freehand branch code partly implements what you would like to see
in the tasking manager.
If I remember well, the idea was to let contributors draw the area they
want to work on instead of having predefined tiles in a grid.

On Tue, Aug 13, 2013 at 4:24 AM, Mikel Maron notifications@github.comwrote:

> Pierre, how is this possibly related to the Freehand branch?
> https://github.com/hotosm/osm-tasking-manager/tree/freehand
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/hotosm/osm-tasking-manager/issues/142#issuecomment-22539550
> .

## 

Pierre GIRAUD
Géomaticien, Analyste

Camptocamp France SAS
Savoie Technolac, BP 352
73377 Le Bourget du Lac, Cedex

Tel : 00 33 4 79 44 44 93
Mail : pierre.giraud@camptocamp.com
http://www.camptocamp.com

**Comment by mikelmaron** [comment]
But does it define the tasks as grids over drawn geometries? Or as arbitrary polygons?

Has this branch been incorporated into the version running on dev?

I wonder if another strategy would be all based on uploading GeoJSON, rather than incorporating drawing tools directly. ie use http://geojson.io/. When uploading, can choose to use the polygons themselves (as the case of the dcbuiling import) or tile the polygons, as the current relation behavior handles.

**Comment by pgiraud** [comment]
About the postGIS support, this is something that would definitely be useful.
FYI, I started a while ago a version 2 (prototype) of the tasking manager which relies on postGIS and Mapnik.

![screenshot8](https://f.cloud.github.com/assets/319774/961925/f9609ef2-04e4-11e3-8936-25fc704b060e.png)
![screenshot9](https://f.cloud.github.com/assets/319774/961927/fbc75816-04e4-11e3-96a0-8c9613414a28.png)

https://github.com/pgiraud/osm-tasking-manager2
